packages/flutter_tools/lib/src/commands/migrate_status.dart

```diff
@@ -0,0 +1,163 @@
+    final String? projectDirectory = stringArg('project-directory');
+    final FlutterProjectFactory flutterProjectFactory = FlutterProjectFactory(logger: logger, fileSystem: fileSystem);
+    final FlutterProject project = projectDirectory == null ? FlutterProject.current() : flutterProjectFactory.fromDirectory(fileSystem.directory(projectDirectory));
+    Directory workingDirectory = project.directory.childDirectory(kDefaultMigrateWorkingDirectoryName);
+    final String? customWorkingDirectoryPath = stringArg('working-directory');
+    if (customWorkingDirectoryPath != null) {
+      if (customWorkingDirectoryPath.startsWith(fileSystem.path.separator) || customWorkingDirectoryPath.startsWith('/')) {
+        // Is an absolute path
+        workingDirectory = fileSystem.directory(customWorkingDirectoryPath);
+      } else {
+        workingDirectory = project.directory.childDirectory(customWorkingDirectoryPath);
+      }
+    }
+    if (!workingDirectory.existsSync()) {
+      logger.printStatus('No migration in progress. Start a new migration with:');
+      printCommandText('flutter migrate start', logger);
+      return const FlutterCommandResult(ExitStatus.fail);
+    }
+
+    final File manifestFile = MigrateManifest.getManifestFileFromDirectory(workingDirectory);
+    if (!manifestFile.existsSync()) {
+      logger.printError('No migrate manifest in the migrate working directory at ${workingDirectory.path}. Fix the working directory or abandon and restart the migration.');
+      return const FlutterCommandResult(ExitStatus.fail);
+    }
+    final MigrateManifest manifest = MigrateManifest.fromFile(manifestFile);
+
+    if (boolArg('diff') || _verbose) {
+      if (boolArg('show-added-files') || _verbose) {
+        for (final String localPath in manifest.addedFiles) {
+          logger.printStatus('Newly addded file at $localPath:\n');
+          try {
+            logger.printStatus(workingDirectory.childFile(localPath).readAsStringSync(), color: TerminalColor.green);
+          } on FileSystemException {
+            logger.printStatus('Contents are byte data\n', color: TerminalColor.grey);
+          }
+        }
+      }
+      final List<String> files = <String>[];
+      files.addAll(manifest.mergedFiles);
+      files.addAll(manifest.resolvedConflictFiles(workingDirectory));
+      files.addAll(manifest.remainingConflictFiles(workingDirectory));
+      for (final String localPath in files) {
+        final DiffResult result = await migrateUtils.diffFiles(project.directory.childFile(localPath), workingDirectory.childFile(localPath));
+        if (result.diff != '' && result.diff != null) {
+          // Print with different colors for better visibility.
+          int lineNumber = -1;
+          for (final String line in result.diff!.split('\n')) {
+            lineNumber++;
+            if (line.startsWith('---') || line.startsWith('+++') || line.startsWith('&&') || _initialDiffLines.contains(lineNumber)) {
```

Comment:
what happens if a line starts with `---`, `+++`, `&&` in the original file?

Reply:
The first character of the line is reserved for the + and - character, so lines that start with those strings in the original files would appear as ` ---` ` +++` and ` &&`. The initial space character would cause the startsWith check to fail.